src/styles/ThemeContext/themeProvider.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState, FC } from 'react';
+import ThemeContext from './themeContext';
+
+
+const ThemeProvider: FC = ({ children }) => {
+
+    const [theme, setTheme] = useState("light");
+
+    const toggleTheme = () => {
+        // change body className
+        let currentTheme = document.body.className;
+        let newTheme = currentTheme === 'light-theme' ? 'dark-theme' : 'light-theme';
+        document.body.className! = newTheme
+        // update state
+        setTheme(theme === 'light' ? 'dark' : 'light');
+    };
+
+    useEffect(() => {
+        // sync theme state with localStorage on first render
+        const localTheme = localStorage.getItem('theme');
+        if (localTheme) {
+            setTheme(localTheme);
+        }
+        // if dark, toggle theme to light on first render
+        if (localTheme === "dark") {
+            document.body.className! = 'dark-theme';
+        }
+    }, []);
```

Comment:
we want to get the theme _before_ the initial render, otherwise we get the flash of the default theme before it updates.

Reply:
Good point, looks like this can just be resolved by removing the `useEffect`, and using the `localStorage` value for the `useState`'s initial value.